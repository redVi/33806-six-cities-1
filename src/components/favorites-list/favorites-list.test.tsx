import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import FavoritesList from "@/components/favorites-list/favorites-list";

const renderer = new ShallowRenderer();

const items = {
  Amsterdam: [
    {id: 1, title: "Amster Hotel", rating: 3.5, price: 250}
  ],
  Paris: [
    {id: 2, title: "Paris Hotel", rating: 4.2, price: 98},
    {id: 3, title: "Paris Hotel 1", rating: 2.3, price: 100}
  ],
}

describe("FavoritesList", () => {
  it("renders correctly", () => {
    renderer.render(<FavoritesList items={items} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
