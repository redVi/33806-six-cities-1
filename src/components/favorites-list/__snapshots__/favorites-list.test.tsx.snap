// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList renders correctly 1`] = `
<React.Fragment>
  <h1
    className="favorites__title"
  >
    Saved listing
  </h1>
  <ul
    className="favorites__list"
  >
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href=""
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </div>
      <PlacesList
        className="favorites__places"
        onImageClick={[Function]}
        places={
          Array [
            Object {
              "id": 1,
              "price": 250,
              "rating": 3.5,
              "title": "Amster Hotel",
            },
          ]
        }
      />
    </li>
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href=""
          >
            <span>
              Paris
            </span>
          </a>
        </div>
      </div>
      <PlacesList
        className="favorites__places"
        onImageClick={[Function]}
        places={
          Array [
            Object {
              "id": 2,
              "price": 98,
              "rating": 4.2,
              "title": "Paris Hotel",
            },
            Object {
              "id": 3,
              "price": 100,
              "rating": 2.3,
              "title": "Paris Hotel 1",
            },
          ]
        }
      />
    </li>
  </ul>
</React.Fragment>
`;
